pkg/controller/controller.go

```diff
@@ -209,6 +221,48 @@ func createWorker(queue workqueue.RateLimitingInterface, resourceType string, ma
 	}()
 }
 
+func (c *controller) monitorConfigMap() {
+	// Cannot wait for the informer to push something into a queue.
+	// What we're waiting on may never exist without us configuring
+	// it, so we have to poll/ ask for it the first time to get it set.
+
+	// Can we ask 'through' an informer? Is it a writeback cache? I
+	// only ever want to monitor and be notified about one configmap
+	// in a hardcoded place.
+	glog.V(9).Info("cluster ID monitor loop enter")
+	cm, err := c.kubeClient.CoreV1().ConfigMaps("default").Get("cluster-info", metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		m := make(map[string]string)
+		m["id"] = c.clusterID
+		cm := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "cluster-info",
+			},
+			Data: m,
+		}
+		_, err := c.kubeClient.CoreV1().ConfigMaps("default").Create(cm)
+		// if we fail to set the id,
+		// it could be due to permissions
+		// or due to being already set while we were trying
+		if nil != err {
+			glog.Warning("could not set clusterid configmap", cm)
+		}
+	} else {
```

Comment:
Need to check for `err != nil` here.

Reply:
Agreed, or rework it so there is a check for err != nil, then inside check if if the error is not found, else log the error; else do the work for happy path.

Reply:
reworked.